src/main/java/com/zoomdata/connector/example/provider/cratedb/CrateDBMetaFlagsDetector.java

```diff
@@ -0,0 +1,107 @@
+import java.util.StringJoiner;
+import java.util.function.Predicate;
+
+import static com.zoomdata.connector.example.common.utils.FieldMetaFlag.PLAYABLE;
+import static com.zoomdata.connector.example.common.utils.FieldMetaFlag.addFlags;
+import static com.zoomdata.connector.example.framework.common.PropertiesExtractor.tableName;
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toSet;
+import static java.util.stream.Stream.of;
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.left;
+import static org.apache.commons.lang3.StringUtils.substringBetween;
+
+@Slf4j
+public class CrateDBMetaFlagsDetector {
+
+    public static final String SHOW_CREATE_TABLE = "show create table ";
+
+    private final SQLTemplates sqlTemplates;
+
+    public CrateDBMetaFlagsDetector(SQLTemplates sqlTemplates) {
+        this.sqlTemplates = sqlTemplates;
+    }
+
+    public void populateMetaFlags(Connection connection, CollectionInfo collectionInfo, List<FieldMetadata> metadata) {
+        getCreateTableStatement(connection, collectionInfo)
+                .map(this::extractPartitionedFields)
+                .ifPresent(partitionedFields -> assignMetaFlagsToFields(metadata, partitionedFields));
+    }
+
+    private Optional<String> getCreateTableStatement(Connection connection, CollectionInfo collectionInfo) {
+        try (Statement ps = connection.createStatement();
+             ResultSet rs = ps.executeQuery(showStatement(collectionInfo))) {
+            return Optional.ofNullable(extractResultSet(rs));
+        } catch (SQLException e) {
+            log.warn("Failed to obtain CREATE TABLE statement for the schema [{}] and collection [{}]",
+                    collectionInfo.getSchema(), collectionInfo.getCollection(), e);
+            return Optional.empty();
+        }
+    }
+
+    private String showStatement(CollectionInfo collectionInfo) {
+        return SHOW_CREATE_TABLE + tableName(collectionInfo, sqlTemplates);
+    }
+
+    private String extractResultSet(ResultSet resultSet) throws SQLException {
+        StringJoiner joiner = new StringJoiner(" ");
+        while(resultSet.next()) {
+            joiner.add(resultSet.getString(1));
+        }
+        return joiner.toString();
+    }
+
+    /**
+     * Extracts names of partitioned fields from the PARTITIONED BY clause of a create statement. For example:
+     * <pre>
+     * PARTITIONED BY (
+     *      first_partition_key INT COMMENT 'The comment',
+     *      another_partition_key INT
+     * )
+     * </pre>
+     */
+    private Set<String> extractPartitionedFields(String createStatement) {
+        String partitionedByClause = ofNullable(substringBetween(createStatement, "PARTITIONED BY (", ")")).orElse(EMPTY);
+        return of(partitionedByClause.split(","))
+                .map(String::trim)
+                .map(this::extractColumnName)
+                .map(String::toLowerCase)
```

Comment:
If the column name is a quoted identifier, then it must be treated as a case-sensitive string. For all other cases, it is fine to lowercase column names. Therefore, we probably should not lowercase all column names.
```
"Foo" -> Foo
Foo -> foo
```
https://crate.io/docs/reference/sql/reference/lexical_structure.html